src/compiler/generate-tests-helper.xsl

```diff
@@ -70,6 +73,12 @@
     <xsl:sequence select="@as" />
 
     <xsl:choose>
+      <xsl:when test="$variable_is_pending">
+        <!-- Define variable, but do not give it a value because the value specified in test file might not be executable. -->
+        <xsl:element name="xsl:{$type}">

```

Comment:
The element `xsl:{$type}` is created inside another `xsl:{$type}`. Looks suspicious.

Also this suspicious code could have been detected if one of the pending `x:variable` instances in `Nonsense variable definition in pending scenario does not cause build failure.` scenario in `xspec-variable.xspec` had `@as`.

Reply:
@AirQuick 
Indeed. Thanks for catching this. Should be fixed in 428a7116e8a7072f377bd9cd01dacfdb2736426d.